src/linechart.ts

```diff
@@ -1,89 +1,75 @@
-//             }, {
-//                 defs: new Defs()
-//             }, {
-//                 legend: new Legend()
-//             }, {
-//                 xScale: new Scale('x')
-//             }, {
-//                 yScale: new Scale('y')
-//             }, {
-//                 y2Scale: new Scale('y2')
-//             }, {
-//                 xAxis: new Axis('x')
-//             }, {
-//                 yAxis: new Axis('y')
-//             }, {
-//                 y2Axis: new Axis('y2')
-//             }, {
-//                 xGrid: new Grid('x')
-//             }, {
-//                 yGrid: new Grid('y')
-//             }, {
-//                 dots: new Dots()
-//             }, {
-//                 lines: new Lines()
-//             }, {
-//                 areas: new Areas()
-//             }, {
-//                 columns: new Columns()
-//             }, {
-//                 stacks: new Stacks()
-//             }, {
-//                 tooltip: new Tooltip()
-//             }, {
-//                 drag: new Drag()
-//             }, {
-//                 zoom: new Zoom()
-//             }, {
-//                 brush: new Brush()
-//             }
-//         ]);
-//         mgr.all().map(function(_plg, key) {
-//             return _plg.init(key, mgr, dis);
-//         });
-//         dis.create();
-//         scope.$watch("data, options", function() {
-//             return dis.update(scope.options);
-//         });
-//         return scope.$on("$destroy", function() {
-//             return dis.exit();
-//         });
-//     };
-//     return {
-//         replace: true,
-//         restrict: 'E',
-//         scope: {
-//             data: '=',
-//             options: '='
-//         },
-//         template: '<div></div>',
-//         link: link
-//     };
-// });
-
-// // ---
-// // generated by coffee-script 1.9.2
 
+    link = (scope: ILineChartScope, element: JQuery, attributes: ng.IAttributes) => {
+
+      // Initialize global events
+      this.em.init([
+        'create',  // on creation of the chart
+        'update',  // on update of the chart
+        'destroy', // on destroying the chart
+        'hover',   // on hover over a data point or column
+        'click',   // on click on a data point or column
+        'focus',   // on focus of a data point from a snappy tooltip
+        'toggle',  // on toggling series' visibility
+      ]);
+
+      // Register all factories
+      // Note: we can apply additional arguments to each factory
+      this.fm.register('container', Factory.Container, element[0]);
+      this.fm.register('x-axis', Factory.Axis, 'x');
+      this.fm.register('y-axis', Factory.Axis, 'y');
+
+      // Initialize all factories
+      this.fm.all().map((f) => f.instance.init(f.key, this.em, this.fm));
+
+      // Trigger the create event
+      this.em.trigger('create');
+
+      // Trigger the update event
+      scope.$watch('options', () => this.em.trigger('update', scope.options), true);
```

Comment:
exactly. actually we should just trigger one update event and listen for both changes. And I guess we should pass a copy of scope.options to avoid an infinite digest when parsing them